lib/consul/async/process_handler.rb

```diff
@@ -9,6 +9,14 @@ class ProcessDoesNotExist < StandardError
     class ProcessHandler
       attr_reader :command, :sig_reload, :sig_term, :pid, :exit_status, :last_signal_sent, :reload_scheduled
       attr_writer :reload_scheduled
+
+      def restart
+        warn "Restart process with pid #{pid}"
+        kill
```

Comment:
There are a few things to consider:

1. restart is using sig_term parameter -> so options to change it will alter its behavior (should be in the help IMHO)
2. killing the process might fail -> This is the case for instance when the process cancels its terminaison by trapping TERM and does not want to stop. In such case, consul-templaterb will likely be stuck forever, waiting for its child process. A possible solution would be to wait for it for a few seconds, then trying with SIGKILL, not sure on how you plan t use it. Anyhow, it should probably be clear in the documentation

Reply:
I agree, managing the process is complicated. I cancel this review, and I will use supervisord !

Reply:
@cpaillet another way would be to use the syntax -t: `-t erb_file:[output]:[command]`

=> with such mode, for instance, you can do:

`-t myconfig.erb:/opt/myprogram/etc/myconfig.cfg:/opt/myprogram/bin/restart.sh`

=> which would launch `/opt/myprogram/bin/restart.sh` each time the file `/opt/myprogram/etc/myconfig.cfg` is being modified... not sure if it help, but it allows for complicated reload/restart behavior